### PR TITLE
[Nala]: Remove redundant overrides, to improve theming support.

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/40689

**Notes:**
- You can still install the old solid color themes from https://chromewebstore.google.com/collection/chrome_color_themes
- I can separate the tri-color theme from this PR, but it'll be easier for @aguscruiz to review with it in for now. 
- The dark theme is slightly less dark after these changes
- If people have a solid color theme enabled, this won't affect it.

I'm not sure whether to merge this as three separate PRs
1. Remove Chromium overrides
2. Enable Tri-Color picker
3. Change default theme

or as one. For now, the three commits are in one PR so @aguscruiz can more easily see how all the changes are going to work together.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. If you select a dual color theme and then upgrade to this PR, the theme should still be active
2. After this PR, only tri-color themes should be visible in brave://settings/manageProfile
3. The new themes should have a light/dark mode (which are different, unlike the old dual color themes)
4. The new light/dark theme should be largely similar to the existing light/dark theme
5. Vertical tabs should look fine in light/dark mode
6. Horizontal tabs should look fine in light/dark mode
7. Omnibox should still have hover/selected state when open
8. If you're editing text in the omnibox and close it, the omnibox should look the same as when its just showing the URL (unlike in Chromium)

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
@simonhong mind taking a look at the color changes? I'll go over this with @aguscruiz tomorrow

**Comment by simonhong** [comment]
This happened again :) 
image

File: chromium_src/ui/color/color_mixers.cc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // You can obtain one at https://mozilla.org/MPL/2.0/.
 
 #include "brave/ui/color//nala/nala_color_mixer.h"
-#include "brave/ui/color/missing_color_ref_mixer.h"
+#include "brave/ui/color/brave_ref_color_mixer.h"
```
Comment (simonhong) [inline chromium_src/ui/color/color_mixers.cc:7]: Below `AddMissingRefColorMixerForNala` should be changed to `AddBraveRefColorMixer`

**Comment by aguscruiz** [comment]
> This happened again :) image
> 
> and different border color around side panel image and vertical tab also has border color. image
> 
> Different icon color of activated toolbar(also active side panel item) image

For the Sidebar issue, having an outline, this is a place where we differ from chromium. 
We don't have a divider between the content and the toolbar. We also don't have a divider between sidebar and content.

I need to see if we want to add an outline in the panel content's frame, to separate them from the chrome background (which is now gonna be the same color). But let me get back to you on that. In the meantime, we should remove those dividers. They won't work with our window padding

image

File: chromium_src/ui/color/ref_color_mixer.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_CHROMIUM_SRC_UI_COLOR_REF_COLOR_MIXER_H_
+#define BRAVE_CHROMIUM_SRC_UI_COLOR_REF_COLOR_MIXER_H_
+
+#include "src/ui/color/ref_color_mixer.h"  // IWYU pragma: export
+
+namespace ui {
+
+void AddGeneratedPalette(ColorProvider* provider,
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply chromium_src/ui/color/ref_color_mixer.h:13]: removed this file

File: chromium_src/ui/color/color_mixers.cc [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply chromium_src/ui/color/color_mixers.cc:7]: removed refs to mixers form the brave_ref_color_mixer

**Comment by fallaciousreasoning** [comment]
@aguscruiz we have a separator between vertical tabs and the content area still - seems like we have this in release too but its a bit more subtle:
image

Fixed separator @simonhong 
image

@aguscruiz I think I've improved the dark mode with a pretty minimal change, which hopefully @rebron will like:
image
image
image
image


Hover states are pretty subtle, unfortunately (see last screenshot)

**Comment by simonhong** [comment]
Seeing some regressions from side panel's header background color.
In private window with light mode,
image

In dark mode, header color & bg colors are different.
image

I think we need to set bg color explicitely to `ReadLaterSidePanelHeaderView` and `BookmarksSidePanelHeaderView`.

and playlist bg color is different with above others. Not sure which color should be used in the panel.
image

**Comment by fallaciousreasoning** [comment]
@simonhong I'm going to fix those in a separate PR, because I think that will need to be uplifted to 1.70.x
https://github.com/brave/brave-browser/issues/40749

**Comment by simonhong** [comment]
> @simonhong I'm going to fix those in a separate PR, ~because I think that will need to be uplifted to 1.70.x~ [brave/brave-browser#40749](https://github.com/brave/brave-browser/issues/40749)
> 
> just checked, it's not in 1.70.x but it is in current Nightly

Oops, thanks for checking. I thought it happens with this PR.

**Review by simonhong — APPROVED** [review]
++

File: chromium_src/ui/color/sys_color_mixer.cc [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "ui/color/sys_color_mixer.h"
+
+#include "brave/ui/color/brave_sys_color_mixer.h"
+
+#define kGrayscale kGrayscale) {                        \
```
Comment (goodov) [inline chromium_src/ui/color/sys_color_mixer.cc:10]: I assume all these overrides has unit tests to catch overrides breakage. If not, then we should add some unit tests for most obvious cases.

File: chromium_src/chrome/browser/ui/views/location_bar/location_bar_view.cc [hunk 5]
```diff
@@ -41,9 +41,11 @@
 #define PageActionIconContainerView BravePageActionIconContainerView
 #define STYLE_BODY_4_EMPHASIS STYLE_PRIMARY
 
-// We don't use different hover color when omnibox doesn't have focus but still
+// We don't use different colors when the omnibox doesn't have focus but still
 // contains in-progress user input.
-#define kColorOmniboxResultsBackgroundHovered kColorLocationBarBackground
+#define kColorOmniboxResultsBackgroundHovered kColorLocationBarBackgroundHovered); \
```
Comment (goodov) [inline chromium_src/chrome/browser/ui/views/location_bar/location_bar_view.cc:46]: is this override covered with a test?

File: chromium_src/ui/color/sys_color_mixer.cc [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    AddGrayscaleSysColorOverrides(mixer, key);          \
+    ui::AddBraveGrayscaleSysColorOverrides(mixer, key); \
+  }                                                     \
+  else if (false
```
Comment (goodov) [inline chromium_src/ui/color/sys_color_mixer.cc:14]: I think it's better to override it via `#define AddSysColorMixer AddSysColorMixer_ChromiumImpl` so the override can be a bit simpler. Is it possible?

```
void AddSysColorMixer(ColorProvider* provider, const ColorProviderKey& key) {
  AddSysColorMixer_ChromiumImpl(provider, key);
  if (key.user_color_source == ColorProviderKey::UserColorSource::kGrayscale) {
    AddBraveGrayscaleSysColorOverrides(mixer, key);
  }
}
```

Comment (fallaciousreasoning) [reply chromium_src/ui/color/sys_color_mixer.cc:14]: I'd actually assumed this wouldn't work (as we'd be putting the overrides in a separate color mixer). However, it seems to turn out fine. 

Changed.

File: chromium_src/chrome/browser/ui/views/location_bar/location_bar_view.cc (hunk 5 above)
Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/location_bar/location_bar_view.cc:46]: Added a test

File: chromium_src/ui/color/BUILD.gn [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+source_set("unit_tests") {
```
Comment (goodov) [inline chromium_src/ui/color/BUILD.gn:6]: I see you implemented these tests in `//brave/ui/color`, which is correct!

this file looks like a leftover.

File: chromium_src/ui/color/sys_color_mixer.cc [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+#include "ui/color/sys_color_mixer.h"
+
+#include "brave/ui/color/brave_sys_color_mixer.h"
+
+#define AddSysColorMixer AddSysColorMixer_Chromium
+// #define kGrayscale kGrayscale) {                        \
+//     AddGrayscaleSysColorOverrides(mixer, key);          \
+//     ui::AddBraveGrayscaleSysColorOverrides(mixer, key); \
+//   }                                                     \
+//   else if (false
```
Comment (goodov) [inline chromium_src/ui/color/sys_color_mixer.cc:15]: leftover to cleanup

File: ui/color/brave_sys_color_mixer.h [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_UI_COLOR_BRAVE_SYS_COLOR_MIXER_H_
+#define BRAVE_UI_COLOR_BRAVE_SYS_COLOR_MIXER_H_
+
+#include "ui/color/color_provider_key.h"
+
+namespace ui {
+class ColorProvider;
+
```
Comment (goodov) [inline ui/color/brave_sys_color_mixer.h:13]: nit: can `ColorProviderKey` also be forwarded?

**Review by goodov — APPROVED** [review]
lgtm with few things to cleanup. nice work!

File: chromium_src/ui/color/BUILD.gn (hunk 7 above)
Comment (fallaciousreasoning) [reply chromium_src/ui/color/BUILD.gn:6]: Whoops, nice catch!

**Comment by fallaciousreasoning** [comment]
Thanks for all your help @simonhong & @goodov :smile: @aguscruiz could you take another look at the PR and see if the SplitView changes look good?

**Comment by aguscruiz** [comment]
Just checked on Windows, all colors are looking great, seal of approval on my end to test this out on nightly and see how it feels :D

**Pull request merged**